ea-2022-04-ndvi-automation.ipynb

```diff
@@ -0,0 +1,1600 @@
+   "metadata": {},
+   "source": [
+    "# Workflow Pseudocode\n",
+    "\n",
+    "1. Open data for single landsat scene/date:\n",
+    "    * Get list of .tif files in folder for single scene/date with glob.\n",
+    "    * Subset list to grab only bands 4 and 5 (those needed to calculate NDVI).\n",
+    "    * Sort file list. \n",
+    "    * Open and crop bands with open_rasterio using shp file in data directory. \n",
+    "2. Calculate average NDVI value for single landsat scene/date. \n",
+    "\n",
+    "3. Get list of all landsat dates/scenes folders for one site using glob.\n",
+    "4. Use steps 1 and 2 above to get average NDVI value for each scene. \n",
+    "    * Grab date and site from file name. \n",
+    "5. Save NDVI, date, and site for each date/scene to pandas df. \n",
+    "6. Export pandas dataframe with mean_ndvi, site, and date to csv file. \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "deletable": false,
+    "editable": false,
+    "nbgrader": {
+     "cell_type": "code",
+     "checksum": "9c7cd3e2e5089092e06ba301f2719a63",
+     "grade": false,
+     "grade_id": "core-imports",
+     "locked": true,
+     "schema_version": 3,
+     "solution": false,
+     "task": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Autograding imports - do not modify this cell\n",
+    "import matplotcheck.autograde as ag\n",
+    "import matplotcheck.notebook as nb\n",
+    "import matplotcheck.timeseries as ts\n",
+    "from datetime import datetime"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "deletable": false,
+    "nbgrader": {
+     "cell_type": "code",
+     "checksum": "3c4d1141999885a9a9b09772962b180a",
+     "grade": true,
+     "grade_id": "student-imports-answer",
+     "locked": false,
+     "points": 10,
+     "schema_version": 3,
+     "solution": true,
+     "task": false
+    },
+    "tags": [
+     "hide",
+     "hide_output"
+    ]
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "This directory exists, and is set to current working directory.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Import needed packages in PEP 8 order\n",
+    "# and no unused imports listed (10 points total)\n",
+    "\n",
+    "# Import necessary libraries\n",
+    "import os\n",
+    "from glob import glob\n",
+    "import shutil\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "from matplotlib import patches as mpatches, colors\n",
+    "from datetime import datetime\n",
+    "import pandas as pd\n",
+    "import geopandas as gpd\n",
+    "import seaborn as sns\n",
+    "import numpy as np\n",
+    "from numpy import ma\n",
+    "import xarray as xr\n",
+    "import rioxarray as rxr\n",
+    "import earthpy as et\n",
+    "import earthpy.plot as ep\n",
+    "import earthpy.mask as em\n",
+    "import earthpy.spatial as es\n",
+    "\n",
```

Comment:
Looks like all of the necessary modules are here and used! Nice work!

Reply:
Thank you!